packages/shared-schemas/src/database-api.schema.ts

```
@@ -275,29 +258,63 @@ export const adminTableRecordsListQuerySchema = z
+// validated structurally, e.g. { "pkKeys": {"tenant_id":"t1","item_id":"i2"}, "data": {...} }.
+export const adminTableRecordUpdateRequestSchema = z.object({
+  pkKeys: adminTableRecordPrimaryKeySchema,
+  data: adminTableRecordUpdateDataSchema,
+});
+
+// Delete request body: one primary-key tuple per record to delete,
+// e.g. { "pkKeys": [{"tenant_id":"t1","item_id":"i2"},{"tenant_id":"t1","item_id":"i3"}] }.
+export const adminTableRecordsDeleteRequestSchema = z.object({
+  pkKeys: z
+    .array(adminTableRecordPrimaryKeySchema)
+    .min(1, 'At least one primary key is required'),
```

ESLint

[error] Replace `⏎····.array(adminTableRecordPrimaryKeySchema)⏎····` with `.array(adminTableRecordPrimaryKeySchema)`

(prettier/prettier)